cmake/compilers/Clang.cmake

```diff
@@ -12,6 +12,12 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+if(EMSCRIPTEN)
+  set(TBB_EMSCRIPTEN 1)
+  set(CMAKE_CXX_FLAGS ${CMAKE_CXX_FLAGS} " -pthread -sINITIAL_MEMORY=65536000 -s ALLOW_MEMORY_GROWTH=1 -s EXIT_RUNTIME=1 -s PROXY_TO_PTHREAD=1 -fexceptions")
+  set(CMAKE_CXX_LINK_FLAGS ${CMAKE_CXX_LINK_FLAGS} -pthread)
```

Comment:
A few questions here:

1. Why do we need to explicitly set `-pthread` here? We already compile and link with `Threads::Threads` which is pthread on Linux.
2. Can we override `TBB_COMMON_COMPILE_FLAGS` and `TBB_COMMON_LINK_FLAGS` instead of these?